src/internal_modules/roc_core/mov_histogram.h

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2023 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+//! @file roc_core/mov_histogram.h
+//! @brief Rolling window moving histogram.
+
+#ifndef ROC_CORE_MOV_HISTOGRAM_H_
+#define ROC_CORE_MOV_HISTOGRAM_H_
+
+#include "roc_core/array.h"
+#include "roc_core/iarena.h"
+#include "roc_core/panic.h"
+#include "roc_core/ring_queue.h"
+
+namespace roc {
+namespace core {
+
+/// @brief
+//! Efficiently implements moving histogram based on approach
+/// @tparam T
+template <typename T> class MovHistogram {
+public:
+    //! Initialize.
+    MovHistogram(IArena& arena, T value_range, size_t num_bins, size_t window_length)
+        : value_range_(value_range)
+        , num_bins_(num_bins)
+        , window_length_(window_length)
+        , bin_width(static_cast<T>(value_range) / static_cast<T>(num_bins))
+        , ring_buffer_(arena, window_length)
+        , bins_(arena)
+        , valid_(false) {
+        if (num_bins == 0 || window_length_ == 0) {
```

Comment:
it would be great not to mix parameters and member values in a single expression